hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -2488,7 +2488,11 @@ func machineTemplateBuilders(hcluster *hyperv1.HostedCluster, nodePool *hyperv1.
 		}
 	case hyperv1.KubevirtPlatform:
 		template = &capikubevirt.KubevirtMachineTemplate{}
-		machineTemplateSpec = kubevirt.MachineTemplateSpec(nodePool, kubevirtBootImage, hcluster)
+		var err error
+		machineTemplateSpec, err = kubevirt.MachineTemplateSpec(nodePool, kubevirtBootImage, hcluster)
+		if err != nil {
+			return nil, nil, "", err
+		}
```

Comment:
we need this error to bubble up to a condition somewhere so we have visibility that the controller is blocked due to json patch errors.

Reply:
I tracked this down a bit. We do get a ReconsileError event associated with the nodepool if this fails, which I think is good enough for now. It at least means we don't have to look at a log message to find when this issue is occurring. 

```
r.recorder.Eventf(nodePool, corev1.EventTypeWarning, "ReconcileError", "%v", err)
```